src/business-logic/services/device-service-implementation.ts

```diff
@@ -0,0 +1,80 @@
+import { DeviceService } from './interfaces/device-service'
+import { MongoDeviceRepository } from '../../data-access/repositories/mongo/mongo-device-repository'
+import { Device } from '../../model/entities/device'
+import { INewsDevice } from '../../model/entities/inews-device'
+import { TelemetricsDevice } from '../../model/entities/telemetrics-device'
+
+export class DeviceServiceImplementation implements DeviceService {
+  constructor(private readonly deviceRepository: MongoDeviceRepository) {}
+  
+  public async readAllConfigurations(): Promise<(Device | INewsDevice | TelemetricsDevice)[]> {
+    return await this.deviceRepository.findAllDevices()
+  }
+
+  public async readConfiguration(deviceId: string): Promise<(Device | INewsDevice | TelemetricsDevice)> {
+    const deviceConfig = await this.deviceRepository.findById(deviceId)
+    if (!deviceConfig) {
+      throw new Error(`Device with id ${deviceId} not found`)
+    }
+    return deviceConfig
+  }
+
+  public async create(_device: Device | INewsDevice | TelemetricsDevice): Promise<void> {
+    const existingDevice = await this.deviceRepository.findById(_device.id)
+    if (existingDevice) {
+      throw new Error(`Device with id ${_device.id} already exists`)
+    }
+    await this.deviceRepository.create(_device)
+  }
+
+  public async update(deviceId: string, config: Device): Promise<void> {
+    const existingDevice = await this.deviceRepository.findById(deviceId)
+    if (!existingDevice) {
+      throw new Error(`Device with id ${deviceId} not found`)
+    }
+    await this.deviceRepository.update(deviceId, config)
+  }
+
+  public async delete(deviceId: string): Promise<void> {
+    const existingDevice = await this.deviceRepository.findById(deviceId)
+    if (!existingDevice) {
+      throw new Error(`Device with id ${deviceId} not found`)
+    }
+    await this.deviceRepository.delete(deviceId)
+  }
+
+  public async disconnect(deviceId: string): Promise<void> {
+    const deviceConfig = await this.deviceRepository.findById(deviceId)
+    if (!deviceConfig) {
+      throw new Error(`Device with id ${deviceId} not found`)
+    }
+    // Simulate disconnecting the device
+    console.log(`Disconnecting device ${deviceId}`)
+    await new Promise((resolve) => setTimeout(resolve, 1000))
+  }
```

Comment:
You don't need to simulate the disconnect (and the connect in the next method) here. You have these methods (or should have) on the Device, so you should just call these methods here and then you can "simulate" it there untill you make the Device implement the real method.
So it would/could look like this:
```
public async disconnect(deviceId: string): Promise<void> {
  const device: Device | undefined = this.deviceRepository.getDevice(deviceId)
  if (!device) {
    return
  }
  await device.disconnect()
}

public async connect(deviceId: string): Promise<void> {
  const device: Device | undefined = this.deviceRepository.getDevice(deviceId)
  if (!device) {
    return
  }
  await device.connect()
}
```
And then in your device class (example: `INewsDevice`):
````
export class INewsDevice extends Device {
  public async connect(): Promise<void> {
    console.log('Connecting to INews Device')
    await new Promise((resolve) => setTimeout(resolve, 1000))
  }

  public async disconnect(): Promise<void> {
    console.log('Disconnecting from INews Device')
    await new Promise((resolve) => setTimeout(resolve, 1000))
  }
}
```